src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,476 @@
+import React, { PureComponent } from "react";
+import compose from "recompose/compose";
+import { StyleSheet, css } from "aphrodite/no-important";
+import * as d3 from "d3";
+import {
+  zoomFunction,
+  distinctColors,
+  transformDataIntoMapArray,
+  getMinMaxValue
+} from "./utils";
+import injectData from "../injectData";
+import exportableToPng from "../exportableToPng";
+
+const styles = StyleSheet.create({
+  divContainer: {
+    overflow: "hidden",
+    position: "relative"
+  },
+  tooltip: {
+    position: "absolute"
+  },
+  vertical: {
+    marginTop: 60,
+    height: 190,
+    pointerEvents: "none"
+  },
+  legend: {
+    position: "relative"
+  },
+  legendItem: {
+    marginTop: 2
+  },
+  legendItemText: {
+    marginLeft: 5
+  },
+  legendButton: {
+    height: 15,
+    width: 15
+  }
+});
+
+class Streamgraph extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: 800,
+      height: 300
+    };
+
+    this.divContainer = React.createRef();
+    this.svgContainer = React.createRef();
+    this.anchor = React.createRef();
+
+    this.xAxisScale;
```

Comment:
What's the point to these declarations?
You don't have to declare them in order to this "this.x" later.

Also, it's really a bad habit to use `this.x` to store values in JS.
Please delete these declarations.

Reply:
I'm somewhat OK with that fact that it doesn't bother React.

But, do you agree with me that this code has no effect and can be removed?

```js
this.xAxisScale;
this.yAxisScale;
this.xAxis;
this.yAxisL;
this.yAxisR;
this.gx;
this.gyr;
this.gyl;
this.streams;
this.hoveredKey;
this.hoveredValue;
```

Reply:
yes absolutely ! I put it there for the understanding of the code but if you think that's useless I delete it, no problems.